ssvsigner/client/client.go

```diff
@@ -0,0 +1,252 @@
+package ssvsignerclient
+
+import (
+	"bytes"
+	"encoding/hex"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+	"time"
+
+	"go.uber.org/zap"
+
+	"github.com/ssvlabs/ssv/ssvsigner/server"
+	"github.com/ssvlabs/ssv/ssvsigner/web3signer"
+)
+
+type Status = server.Status
+
+const (
+	StatusImported   Status = "imported"
+	StatusDuplicated Status = "duplicate"
+	StatusDeleted    Status = "deleted"
+)
+
+type ShareDecryptionError error
+
+type SSVSignerClient struct {
+	logger     *zap.Logger
+	baseURL    string
+	httpClient *http.Client
+}
+
+func New(baseURL string, opts ...Option) *SSVSignerClient {
+	baseURL = strings.TrimRight(baseURL, "/")
+
+	c := &SSVSignerClient{
+		baseURL: baseURL,
+		httpClient: &http.Client{
+			Timeout: 30 * time.Second,
+		},
+	}
+
+	for _, opt := range opts {
+		opt(c)
+	}
+
+	return c
+}
+
+type Option func(*SSVSignerClient)
+
+func WithLogger(logger *zap.Logger) Option {
+	return func(client *SSVSignerClient) {
+		client.logger = logger
+	}
+}
+
+func (c *SSVSignerClient) AddValidators(encryptedPrivKeys ...[]byte) ([]Status, [][]byte, error) {
+	privKeyStrs := make([]string, 0, len(encryptedPrivKeys))
+	for _, privKey := range encryptedPrivKeys {
+		privKeyStrs = append(privKeyStrs, hex.EncodeToString(privKey))
+	}
+
+	req := server.AddValidatorRequest{
+		EncryptedSharePrivateKeys: privKeyStrs,
+	}
+
+	reqBytes, err := json.Marshal(req)
+	if err != nil {
+		return nil, nil, fmt.Errorf("marshal request: %w", err)
+	}
+
+	url := fmt.Sprintf("%s/v1/validators/add", c.baseURL)
+	httpResp, err := c.httpClient.Post(url, "application/json", bytes.NewReader(reqBytes))
+	if err != nil {
+		return nil, nil, fmt.Errorf("request failed: %w", err)
+	}
+	defer func() {
+		if err := httpResp.Body.Close(); err != nil {
+			c.logger.Error("failed to close http response body", zap.Error(err))
+		}
+	}()
+
+	respBytes, err := io.ReadAll(httpResp.Body)
+	if err != nil {
+		return nil, nil, fmt.Errorf("read response body: %w", err)
+	}
+
+	if httpResp.StatusCode != http.StatusOK {
+		if httpResp.StatusCode == http.StatusUnauthorized {
+			return nil, nil, ShareDecryptionError(errors.New(string(respBytes)))
```

Comment:
`errors.Wrap` is from "github.com/pkg/errors", which is deprecated in favor of `fmt.Errorf("%w", err)`

Reply:
I think `errors.New` is preferable for non-wrapped errors but importing both `errors` and `github.com/pkg/error` would look weird. Also, removing `github.com/pkg/error` here would increase a scope of changes, however, maybe I'll choose this way as I move some files

Reply:
No, it's from `"errors"`, not `"github.com/pkg/errors"`. While I'm fine with using `fmt.Errorf` for non-wrapped errors, I think there's no issue in using `errors.New`